src/WikiSearch/WikiSearch.tsx

```diff
@@ -0,0 +1,177 @@
+        .test(searchKey)) && <div key={recentSearch} onClick={() => {
+            const newParam = new URLSearchParams();
+            newParam.set('q', recentSearch);
+            setSearchParams(newParam);
+            setSearchKey(recentSearch);
+        }} className="popover-content">{recentSearch}</div>).filter(Boolean);
+
+    const noResults = !loading && searchKey && searchResults && Array.isArray(searchResults[1]) && searchResults[1].length === 0;
+
+    /**
+     * TODO: Make Scroll Paginated
+     */
+    return <main>
+        <div id={Styles.QwikiSearchContainer}>
+            <Popover placement="bottom-start">
+                <PopoverTrigger>
+                    <textarea
+                        id={Styles.QwikiSearch}
+                        name="qwiki-search"
+                        role="search"
+                        autoFocus
+                        onChange={(e) => {
+                            const newParam = new URLSearchParams();
+                            newParam.set('q', e.currentTarget.value.replace(/\n/g, ''));
+                            setSearchParams(newParam);
+                            setSearchKey(e.currentTarget.value.replace(/\n/g, ''));
+
+                            if (!e.currentTarget.value) setSearchResults(undefined);
+                        }}
+                        placeholder="Seach Wiki Titles"
+                        value={searchKey}
+                        rows={1}
+                    />
+                </PopoverTrigger>
+                {!searchKey && Array.isArray(items) && items.length > 0 && <PopoverContent className="Popover">
+                    <PopoverDescription>
+                        <div className={Styles.Title}>
+                            <h6>Recent Searches</h6> <a href="/#" onClick={() => setRecentSearches([])}><h6>Clear</h6></a>
+                        </div>
+                        {items}
+                    </PopoverDescription>
+                </PopoverContent>}
+            </Popover>
+            <img src={SearchIcon} alt="Search Icon" />
+        </div>
+        <div className={Styles.History}>
+            {(recentSearches?.length ?? 0) > 0 && <>
+                Search History: {recentSearches?.map(item => <div onClick={() => {
+                    const newParam = new URLSearchParams();
+                    newParam.set('q', item);
+                    setSearchParams(newParam);
+                    setSearchKey(item);
+                }}>{item}</div>)}
+            </>}
+        </div>
+        <div id={Styles.QwikiSearchResultContainer} ref={scrollViewer}>
+            <div className={Styles.QwikiSearchResults}>
+                {
+                    searchResults?.[1].map((_, index) => <a key={searchResults[3][index]} target={`wiki-${searchResults[3][index]}`} href={searchResults[3][index]} className={Styles.QwikiSearchResult}>
```

Comment:
again, please use a separate variable for this, nobody knows what `searchResults?.[1]`, `searchResults?.[3]`.
Ideally a better approach would've been to transform the response in a well defined array of objects.